app/models/study.rb

```diff
@@ -81,6 +81,7 @@ def persisted
       available.reject(&:new_record?)
     end
   end
+  accepts_nested_attributes_for :study_files, allow_destroy: true
```

Comment:
interesting, I haven't used allow_destroy before.  Reading up on it, I like the rails approach of making you have to specify the ids of  models to destroy explicitly, rather than the "oh, you accidentally passed the study back with an empty array for study_files?  I'll delete them all" approach of some other frameworks

Reply:
This isn't a change, just a reordering of declarations because of the issue I found with `StudyDetail` objects.